### PR TITLE
[enhancement](plsql) add regression for failure path

### PR DESCRIPTION
**Pull request by Vallishp**
## Proposed changes

Issue Number: close #xxx

add regression for exception scenarios.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowProcedureStatusCommand.java [hunk 1]
```diff
@@ -102,8 +102,8 @@ private void validateAndExtractFilters(StringBuilder dbFilter, StringBuilder pro
             if ((!columnName.toLowerCase().equals(colDb))
                     && (!columnName.toLowerCase().equals(colName))
                     && (!columnName.toLowerCase().equals(colProcName))) {
-                throw new AnalysisException("Only supports filter" + colProcName + ", "
-                        + colName + "," + colProcName + "with equalTo or LIKE");
+                throw new AnalysisException("Only supports filter  " + colProcName + ", "
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowProcedureStatusCommand.java:105]:
```suggestion
                throw new AnalysisException("Only supports filter " + colProcName + ", "
```

File: regression-test/suites/plsql_p2/test_plsql_show_procedure_fail.groovy [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    }
+    try {
+         sql """SET enable_fallback_to_original_planner=false;"""
+         sql """SHOW PROCEDURE STATUS where Db="Test" or name="Test1";"""
+         fail();
+    } catch (Throwable t) {
+         log.info("{}", t.toString())
+         if (t instanceof java.sql.SQLException) {
+             String result = t.toString()
+             String expected = "java.sql.SQLException: errCode = 2, detailMessage = Only support equalTo  and Like filters.";
```
Comment (xinyiZzz) [inline regression-test/suites/plsql_p2/test_plsql_show_procedure_fail.groovy:64]: real error msg is: `only support AND conjunction, not support OR`
can do this first and change it later

**Comment by xinyiZzz** [comment]
run buildall

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**